index.js

```diff
@@ -0,0 +1,122 @@
+const jsonValidator = require('json-dup-key-validator');
+const path = require('path');
+const fs = require('fs');
+const util = require('util');
+
+const { promisify } = util;
+const readDirAsync = promisify(fs.readdir);
+const readFileAsync = promisify(fs.readFile);
+
+const laodFilesNames = async directoryPath => await readDirAsync(directoryPath);
+
+const loadFilesContent = async (filesNames, dir) => {
+  const filesData = [];
+  const duplicatedKeys = [];
+
+  await Promise.all(
+    filesNames.map(async fileName => {
+      if (fileName.endsWith('.json')) {
+        const fileData = await readFileAsync(`./${dir}/${fileName}`, 'utf8');
+        const parsedFileData = await jsonValidator.parse(fileData, true);
+        filesData.push({
+          fileName,
+          fileData: parsedFileData
+        });
+
+        const duplicatedKey = await jsonValidator.validate(fileData, false);
+        if (duplicatedKey) {
+          duplicatedKeys.push({
+            type: '[DUPLICATED]',
+            key: duplicatedKey,
+            fileName: fileName
+          });
+        }
+      }
+    })
+  );
+
+  return [filesData, duplicatedKeys];
+};
+
+const getAllFilesData = async dir => {
+  const directoryPath = path.join(__dirname, dir);
+  const filesNames = await laodFilesNames(directoryPath);
+  return await loadFilesContent(filesNames, dir);
+};
+
+const createKeysList = filesData => {
+  const keysList = [];
+  for (const key in filesData) {
```

Comment:
```filesData``` is an array of objects containing { fileName, fileData } keys, so ```const key``` seems to be a misleading name